Cabal/Distribution/Simple/Haddock.hs

```diff
@@ -185,9 +186,9 @@ haddock pkg_descr lbi suffixes flags' = do
     when (flag haddockQuickJump && version < mkVersion [2,19]) $ do
       let msg = "Haddock prior to 2.19 does not support the --quickjump flag."
           alt = "The generated documentation won't have the QuickJump feature."
-      case haddockTarget of
-        ForDevelopment -> die' verbosity msg
-        ForHackage -> warn verbosity (msg ++ "\n" ++ alt) 
+      if Flag True == quickJmpFlag
```

Comment:
This is not quite what @hvr suggested IIUC. Since `--for-hackage` implies `--quickjump`, all of `--for-hackage`, `--for-hackage --quickjump`, and `--quickjump` will now fail hard when using an older haddock.

Reply:
It is though (I've even tried it out). `quickJmpFlag` records what the quickjump flag was _prior_ to the `--for-hackage` implication. That way, only `--for-hackage --quickjump`, and `--quickjump` will fail hard when using an older haddock - `--for-hackage` by itself is just a warning.

Reply:
Great! Don't we still get a warning when neither `--for-hackage` nor `--quickjump` were specified, but `haddock` is old?

Reply:
@23Skidoo I don't see any warning w/ e.g. GHC 8.2.2 (i.e. old haddock) and a plain `cabal new-haddock -w ghc-8.2.2` w/o any additional flags.

Reply:
OK. Let's merge!